core/main/src/firebolt/handlers/hdmi_rpc.rs

```diff
@@ -0,0 +1,198 @@
+                .state
+                .get_client()
+                .send_extn_request(HdmiRequest::StopHdmiInput(request))
+                .await
+            {
+                if let ExtnPayload::Response(ExtnResponse::Value(value)) = response.payload {
+                    if let Ok(res) = serde_json::from_value::<HdmiSelectOperationResponse>(value) {
+                        return Ok(res);
+                    }
+                }
+            }
+
+            Err(rpc_err("FB error response TBD"))
+        }
+    }
+
+    async fn get_available_inputs(
+        &self,
+        _ctx: CallContext,
+    ) -> RpcResult<GetAvailableInputsResponse> {
+        if let Ok(response) = self
+            .state
+            .get_client()
+            .send_extn_request(HdmiRequest::GetAvailableInputs)
+            .await
+        {
+            if let ExtnPayload::Response(ExtnResponse::Value(value)) = response.payload {
+                if let Ok(res) = serde_json::from_value::<GetAvailableInputsResponse>(value) {
+                    return Ok(res);
+                }
+            }
+        }
+
+        Err(rpc_err("FB error response TBD"))
+    }
+    async fn on_hdmi_connection_changed(
+        &self,
+        ctx: CallContext,
+        request: ListenRequest,
+    ) -> RpcResult<ListenerResponse> {
+        let listen = request.listen;
+
+        AppEvents::add_listener(
+            &self.state,
+            HDMI_CONNECTION_CHANGED.to_string(),
+            ctx.clone(),
+            request,
+        );
+
+        if self
+            .state
+            .get_client()
+            .send_extn_request(DeviceEventRequest {
+                event: DeviceEvent::HdmiConnectionChanged,
+                id: ctx.app_id,
+                subscribe: listen,
+                callback_type: DeviceEventCallback::FireboltAppEvent,
+            })
+            .await
+            .is_err()
+        {
+            error!("Error while registration");
+        }
+
+        Ok(ListenerResponse {
+            listening: listen,
+            event: HDMI_CONNECTION_CHANGED.to_string(),
+        })
+    }
+    async fn on_auto_low_latency_mode_signal_changed(
+        &self,
+        ctx: CallContext,
+        request: ListenRequest,
+    ) -> RpcResult<ListenerResponse> {
+        let listen = request.listen;
+
+        AppEvents::add_listener(
+            &self.state,
+            AUTO_LOW_LATENCY_MODE_SIGNAL_CHANGED.to_string(),
+            ctx.clone(),
+            request,
+        );
+
+        if self
+            .state
+            .get_client()
+            .send_extn_request(DeviceEventRequest {
+                event: DeviceEvent::AutoLowLatencyModeSignalChanged,
+                id: ctx.app_id,
+                subscribe: listen,
+                callback_type: DeviceEventCallback::FireboltAppEvent,
+            })
+            .await
+            .is_err()
+        {
+            error!("Error while registration");
```

Comment:
Should we still return an OK response even if we hit this block?

Reply:
This is similar to how on_hdcp_changed listener is implemented https://github.com/rdkcentral/Ripple/blob/0565eeeb871988ca1d4a8535d83ce9dc50e3d6e6/core/main/src/firebolt/handlers/device_rpc.rs#L417

Can you elaborate on how I should change it?

Reply:
Lets discuss this one here: https://github.com/rdkcentral/Ripple/pull/264/files/e7ce1599db245e580e921ea96e50d548af3a1a8d#r1350384440

and apply the output to both.